**Core -> Full Integration (with SecurityManager)** [Build 15518](https://ci.wildfly.org/buildConfiguration/WildFlyCore_PullRequest_WildFlyCoreFullIntegrationBootableJarJdk21/568684) outcome was **FAILURE** using a merge of abc0094f590d9adcc66951123cbca9d87032a2ab
Summary: Tests failed: 40 (40 new), passed: 4202, ignored: 48 Build time: 03:07:09
### Failed tests
```
org.wildfly.test.integration.microprofile.faulttolerance.context.asynchronous.AsynchronousRequestContextTestCase: org.wildfly.test.integration.microprofile.faulttolerance.context.asynchronous.AsynchronousRequestContextTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 596041 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)


org.wildfly.test.integration.microprofile.faulttolerance.context.timeout.TimeoutContextTestCase: org.wildfly.test.integration.microprofile.faulttolerance.context.timeout.TimeoutContextTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 596318 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)


org.wildfly.test.integration.microprofile.faulttolerance.multideployment.MultipleDeploymentMetricsTestCase: org.wildfly.test.integration.microprofile.faulttolerance.multideployment.MultipleDeploymentMetricsTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 596475 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)


org.wildfly.test.integration.microprofile.faulttolerance.opentelemetry.FaultToleranceOpenTelemetryIntegrationTestCase: org.wildfly.test.integration.microprofile.faulttolerance.opentelemetry.FaultToleranceOpenTelemetryIntegrationTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 596557 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)


org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationLiveHTTPEndpointTestCase: org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationLiveHTTPEndpointTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 596645 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)


org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationLiveOperationTestCase: org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationLiveOperationTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 596749 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)


org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationReadyHTTPEndpointTestCase: org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationReadyHTTPEndpointTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 596904 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)


org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationReadyOperationTestCase: org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationReadyOperationTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 597018 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)


org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationStartupHTTPEndpointTestCase: org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationStartupHTTPEndpointTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 597099 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)


org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationStartupOperationTestCase: org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationStartupOperationTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 597370 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)


org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationWithoutReadinessHTTPEndpointTestCase: org.wildfly.test.integration.microprofile.health.MicroProfileHealthApplicationWithoutReadinessHTTPEndpointTestCase.: 	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:91)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:142)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
Caused by: org.wildfly.plugin.tools.server.ServerManagerException: The process 597485 is no longer active.
	at org.wildfly.plugin.tools.server.AbstractServerManager.waitFor(AbstractServerManager.java:106)
	at org.wildfly.plugin.tools.server.StandaloneManager.waitFor(StandaloneManager.java:23)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:80)
	... 56 more
------- Stdout: -------
Exception in thread "main" java.lang.IllegalStateException: The install directory /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly may still be in the process of being deleted. The marker file /opt/buildAgent/work/e8e0dd9c7c4ba60/full/testsuite/integration/expansion/target/wildfly/wildfly-cleanup-marker has not been deleted within 10s. Please check for a previous job running and delete the marker file if it was left behind because of an error.
	at org.wildfly.core.jar.boot.Main.waitForDelete(Main.java:342)
	at org.wildfly.core.jar.boot.Main.main(Main.java:128)



##### there are 30 more failed tests, see build details
```

--- a/bootable-jar/runtime/pom.xml
+++ b/bootable-jar/runtime/pom.xml
@@ -120,6 +120,32 @@
                         <test.jboss.home>${project.build.directory}</test.jboss.home>
                     </systemPropertyVariables>
                 </configuration>
+                <executions>
+                  <execution>
+                    <id>test-with-env-disabled</id>
+                    <configuration>
+                      <environmentVariables>
+                        <DISABLE_JDK_SERIAL_FILTER>true</DISABLE_JDK_SERIAL_FILTER>
+                      </environmentVariables>
+                    </configuration>
+                    <goals>
+                      <goal>test</goal>
+                    </goals>
+                    <phase>test</phase>
+                  </execution>
+                  <execution>
+                    <id>test-with-env-set</id>
+                    <configuration>
+                      <environmentVariables>
+                        <JDK_SERIAL_FILTER>custom-value</JDK_SERIAL_FILTER>
+                      </environmentVariables>
+                    </configuration>
+                    <goals>
+                      <goal>test</goal>
+                    </goals>
+                    <phase>test</phase>
+                  </execution>
+                </executions>
             </plugin>
         </plugins>
     </build>

--- a/bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/Arguments.java
+++ b/bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/Arguments.java
@@ -8,6 +8,7 @@ import static org.wildfly.core.jar.runtime.Constants.DEPLOYMENT_ARG;
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.ObjectInputFilter;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.nio.file.Files;
@@ -32,6 +33,11 @@ import org.wildfly.core.jar.runtime._private.BootableJarLogger;
  * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
  */
 final class Arguments {
+    private static final String JDK_SERIAL_FILTER = "jdk.serialFilter";
+    private static final String DISABLE_JDK_SERIAL_FILTER_ENV = "DISABLE_JDK_SERIAL_FILTER";
+    private static final String JDK_SERIAL_FILTER_ENV = "JDK_SERIAL_FILTER";
+    // This default value comes from the standalone.conf file, in case it is changes, make sure to update this value.
+    private static final String DEFAULT_SERIAL_FILTER = "maxbytes=10485760;maxdepth=128;maxarray=100000;maxrefs=300000";
 
     private Arguments(final BootableEnvironment environment) {
         this.environment = environment;
@@ -43,6 +49,8 @@ final class Arguments {
     private final BootableEnvironment environment;
     private Path deployment;
     private Path cliScript;
+    private String serialFilter;
+    private boolean logSerialFilterAlreadySet;
 
     static Arguments parseArguments(final List<String> args, final BootableEnvironment environment) throws Exception {
         Objects.requireNonNull(args);
@@ -54,6 +62,7 @@ final class Arguments {
     private void handleArguments(List<String> args) throws Exception {
         final Map<String, String> systemProperties = new HashMap<>();
         final Iterator<String> iter = args.iterator();
+        String jdkSerialFilterSet = null;
         while (iter.hasNext()) {
             final String a = iter.next();
             if (a.startsWith(DEPLOYMENT_ARG)) {
@@ -66,6 +75,7 @@ final class Arguments {
                     final String urlSpec = iter.next();
                     serverArguments.add(urlSpec);
                     addSystemProperties(makeUrl(urlSpec), systemProperties);
+                    jdkSerialFilterSet = systemProperties.get(JDK_SERIAL_FILTER);
                 } else {
                     throw BootableJarLogger.ROOT_LOGGER.invalidArgument(a);
                 }
@@ -74,6 +84,7 @@ final class Arguments {
                 // We need these set as system properties early so the log manager can use them
                 final String urlSpec = parseValue(a, CommandLineConstants.PROPERTIES);
                 addSystemProperties(makeUrl(urlSpec), systemProperties);
+                jdkSerialFilterSet = systemProperties.get(JDK_SERIAL_FILTER);
             } else if (a.startsWith(CommandLineConstants.SECURITY_PROP)) {
                 serverArguments.add(a);
             } else if (a.startsWith(CommandLineConstants.SYS_PROP)) {
@@ -82,6 +93,7 @@ final class Arguments {
                 // we also need to set it as a current system property for usage in the log manager.
                 serverArguments.add(a);
                 addSystemProperty(a, systemProperties);
+                jdkSerialFilterSet = systemProperties.get(JDK_SERIAL_FILTER);
             } else if (a.startsWith(CommandLineConstants.START_MODE)) {
                 serverArguments.add(a);
             } else if (a.startsWith(CommandLineConstants.DEFAULT_MULTICAST_ADDRESS)) {
@@ -105,10 +117,39 @@ final class Arguments {
                 throw BootableJarLogger.ROOT_LOGGER.unknownArgument(a);
             }
         }
+        // Special handling of jdk.serialFilter
+        // The filter could have been set by a system property not handled in the user arguments.
+        // We will know after having check for the existence of a filter.
+        ObjectInputFilter configuredFilter = ObjectInputFilter.Config.getSerialFilter();
+        String filter = System.getenv(JDK_SERIAL_FILTER_ENV);
+        String disabled = System.getenv(DISABLE_JDK_SERIAL_FILTER_ENV);
+        boolean filterEnabled = disabled == null || !disabled.equalsIgnoreCase("true");
+        if (configuredFilter == null) {
+            if (jdkSerialFilterSet == null) {
+                if (filterEnabled) {
+                    if (filter == null) {
+                        filter = DEFAULT_SERIAL_FILTER;
+                    }
+                    serialFilter = filter;
+                }
+            } else {
+                // The filter has been explicitly set, don't keep the system property.
+                // The filter will be set by the Bootable JAR runtime.
+                serialFilter = jdkSerialFilterSet;
+                systemProperties.remove(JDK_SERIAL_FILTER);
+            }
+        } else if (filter != null && filterEnabled) {
+            logSerialFilterAlreadySet = true;
+        }
         // Add the system properties to the environment
         environment.setSystemProperties(systemProperties);
     }
 
+    void logArgumentsHandling(BootableJarLogger log) {
+        if (logSerialFilterAlreadySet) {
+            log.advertiseSerialFilterSet();
+        }
+    }
     private Path checkPath(String path) {
         Path filePath = Paths.get(path);
         if (!Files.exists(filePath)) {
@@ -155,6 +196,14 @@ final class Arguments {
 
     public Path getCLIScript() {
         return cliScript;
+    }
+
+    /**
+     * The serialization filter that must be explicitely set.
+     * @return The filter, null if no filter to set.
+     */
+    public String getRequiredSerialFilter() {
+        return serialFilter;
     }
 
     private static void addSystemProperty(final String arg, final Map<String, String> properties) {

--- a/bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java
+++ b/bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java
@@ -11,6 +11,7 @@ import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.io.ObjectInputFilter;
 import java.io.UncheckedIOException;
 import java.lang.management.ManagementFactory;
 import java.lang.reflect.InvocationTargetException;
@@ -116,6 +117,8 @@ public final class BootableJar implements ShutdownHandler {
 
         // logging needs to be configured before other components have a chance to initialize a logger
         configureLogger();
+        // Some traces to log could have been captured during args handling
+        arguments.logArgumentsHandling(log);
         long t = System.currentTimeMillis();
         if (arguments.getDeployment() != null) {
             setupDeployment(arguments.getDeployment());
@@ -341,6 +344,13 @@ public final class BootableJar implements ShutdownHandler {
         if (arguments.isHelp()) {
             CmdUsage.printUsage(productConfig, System.out);
             return;
+        }
+
+        if (arguments.getRequiredSerialFilter() != null) {
+            ObjectInputFilter objInputFilter = ObjectInputFilter.Config.createFilter(arguments.getRequiredSerialFilter());
+            // If a filter has already been set (it should not), this call will fail
+            // with an IllegalStateException.
+            ObjectInputFilter.Config.setSerialFilter(objInputFilter);
         }
 
         // Side effect is to initialise Log Manager

--- a/bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/_private/BootableJarLogger.java
+++ b/bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/_private/BootableJarLogger.java
@@ -164,4 +164,8 @@ public interface BootableJarLogger extends BasicLogger {
     @LogMessage(level = DEBUG)
     @Message(id = 25, value = "Failed to initialize a security provider. Reason: %s")
     void securityProviderFailed(Throwable ex);
+
+    @LogMessage(level = INFO)
+    @Message(id = 26, value = "The JDK_SERIAL_FILTER environment variable is being ignored because the global ObjectInputFilter was already configured by the jdk.serialFilter system or security property")
+    void advertiseSerialFilterSet();
 }

--- a/bootable-jar/runtime/src/test/java/org/wildfly/core/jar/runtime/ArgumentsTestCase.java
+++ b/bootable-jar/runtime/src/test/java/org/wildfly/core/jar/runtime/ArgumentsTestCase.java
@@ -4,6 +4,7 @@
  */
 package org.wildfly.core.jar.runtime;
 
+import java.io.FileWriter;
 import java.net.URL;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -13,6 +14,7 @@ import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Properties;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
@@ -28,6 +30,7 @@ import org.junit.Test;
  */
 public class ArgumentsTestCase {
 
+    private static final String FILTER_PROP = "jdk.serialFilter";
 
     @Test
     public void test() throws Exception {
@@ -111,6 +114,46 @@ public class ArgumentsTestCase {
                 throw new Exception("Should have failed");
             }
         }
+
+        {
+            String[] args = {};
+            final TestPropertyUpdater propertyUpdater = new TestPropertyUpdater();
+            Arguments arguments = Arguments.parseArguments(Arrays.asList(args), createEnvironment(propertyUpdater));
+            if (System.getenv("DISABLE_JDK_SERIAL_FILTER") == null) {
+                Assert.assertNotNull("Expected filter to exist.", arguments.getRequiredSerialFilter());
+                String val = System.getenv("JDK_SERIAL_FILTER");
+                if (val != null) {
+                    Assert.assertEquals("Expected the value " + val + " for the serial filter", val, arguments.getRequiredSerialFilter());
+                }
+            } else {
+                Assert.assertNull("No filter expected ", arguments.getRequiredSerialFilter());
+            }
+        }
+
+        {
+            Properties p = new Properties();
+            p.setProperty(FILTER_PROP, "foo");
+            Path propsFile = Files.createTempFile(null, ".properties");
+            propsFile.toFile().deleteOnExit();
+            try (FileWriter w = new FileWriter(propsFile.toFile())) {
+                p.store(w, "");
+            }
+            String[] args = {"--properties", propsFile.toString()};
+            final TestPropertyUpdater propertyUpdater = new TestPropertyUpdater();
+            Arguments arguments = Arguments.parseArguments(Arrays.asList(args), createEnvironment(propertyUpdater));
+            // The property is not set, it has been replaced by an explicit call.
+            Assert.assertFalse("Property " + FILTER_PROP + " should not be set.", propertyUpdater.properties.containsKey(FILTER_PROP));
+            Assert.assertEquals("Expected the value foo for the serial filter", "foo", arguments.getRequiredSerialFilter());
+        }
+    }
+
+    @Test
+    public void testSerialFilterProperty() throws Exception {
+        final List<String> args = Collections.singletonList("-D"+FILTER_PROP+"=foo2");
+        final TestPropertyUpdater propertyUpdater = new TestPropertyUpdater();
+        Arguments arguments = Arguments.parseArguments(args, createEnvironment(propertyUpdater));
+        Assert.assertFalse("Property " + FILTER_PROP + " should not be set.", propertyUpdater.properties.containsKey(FILTER_PROP));
+        Assert.assertEquals("Expected the value foo2 for the serial filter", "foo2", arguments.getRequiredSerialFilter());
     }
 
     @Test

